sdk/eventhub/Azure.Messaging.EventHubs/src/Amqp/AmqpConsumer.cs

```diff
@@ -444,6 +508,61 @@ public override async Task CloseAsync(CancellationToken cancellationToken)
             return link;
         }
 
+        /// <summary>
+        ///   Closes the AMQP link used by the consumer, capturing the passive terminal exception that triggered
+        ///   closing if it should be surfaced during the next requested operation.
+        /// </summary>
+        ///
+        protected void CloseConsumerLink(ReceivingAmqpLink link)
+        {
+            // If there is no link, then no action needs to be
+            // taken.
+
+            if (link == null)
+            {
+                return;
+            }
+
+            // If the consumer is being closed, the sentinel variable will already be set, and
+            // the terminal exception need not be considered.
+
+            if (!_closed)
+            {
+                var linkException = GetTerminalException(link);
```

Comment:
nit: this could be simplified to:
```suggestion
                var linkException = link?.TerminalException;
```